src/ui/public/agg_types/__tests__/param_types/_field.js

```diff
@@ -24,4 +27,38 @@ describe('Field', function () {
       expect(aggParam).to.be.a(BaseAggParam);
     });
   });
+
+  describe('getFieldOptions', function () {
+    it('should return only aggregatable fields', function () {
+      const aggParam = new FieldAggParam({
+        name: 'field'
+      });
+
+      const fields = aggParam.getFieldOptions({
+        getIndexPattern: () => indexPattern
+      });
+      for (let field of fields) {
+        expect(field.aggregatable).to.be(true);
+      }
+    });
+
+    it('should return all fields', function () {
+      const aggParam = new FieldAggParam({
+        name: 'field'
+      });
+
+      aggParam.onlyAggregatable = false;
+
+      const fields = aggParam.getFieldOptions({
+        getIndexPattern: () => indexPattern
+      });
+      let nAggregatable = 0;
+      for (let field of fields) {
+        if (field.aggregatable) {
+          nAggregatable++;
+        }
+      }
+      expect(fields.length - nAggregatable > 0).to.be(true);
```

Comment:
You could also simplify this with some lodash methods:

const nonAggregatableFields = _.reject(fields, 'aggregatable');
expect(nonAggregatableFields).to.not.be.empty();